core/server/data/migrations/versions/3.4/01-add-self-signup-and-from address-to-members-settings.js

```diff
@@ -0,0 +1,107 @@
+const _ = require('lodash');
+
+const common = require('../../../../lib/common');
+
+module.exports.config = {
+    transaction: true
+};
+
+const settingsKey = 'members_subscription_settings';
+const complimentaryPlan = {
+    name: 'Complimentary',
+    currency: 'usd',
+    interval: 'year',
+    amount: '0'
+};
+
+module.exports.up = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
```

Comment:
Should we be adding them here?

Reply:
It's a general subscription config that is being checked here :) We are adding a plan which depends on the config already being present

Reply:
This whole migration will be removed in favour of hardcoded plan entry anyway :wink: